### ISSUE TITLE
DataDog API key logged when restarting systemd unit

### ISSUE THREAD

**Issue by pjvelez** [DataDog/dd-agent#3533]
Hey DataDog!

Just restarted my `datadog-agent` systemd unit and noticed that my API key was logged by `dogstatsd.py` upon doing so. Figured I'd let you know.

**Additional environment details (Operating System, Cloud provider, etc):**
CentOS7 on AWS EC2

**Steps to reproduce the issue:**
1. Restart the systemd unit via `systemctl restart datadog-agent`
2. Check datadog logs

**Describe the results you received:**
Datadog logs an error including my API key in plain text:

```
Sep 28 23:21:31 HOSTNAME systemd[1]: Stopping "Datadog Agent"...
Sep 28 23:21:31 HOSTNAME supervisorctl[8767]: Shut down
Sep 28 23:21:31 HOSTNAME dd.dogstatsd[745]: ERROR (dogstatsd.py:338): Unable to post payload.
                                                Traceback (most recent call last):
                                                  File "/opt/datadog-agent/agent/dogstatsd.py", line 328, in submit_http
                                                    r = requests.post(url, data=data, timeout=5, headers=headers)
                                                  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/requests/api.py", line 110, in post
                                                    return request('post', url, data=data, json=json, **kwargs)
                                                  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/requests/api.py", line 56, in request
                                                    return session.request(method=method, url=url, **kwargs)
                                                  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/requests/sessions.py", line 475, in request
                                                    resp = self.send(prep, **send_kwargs)
                                                  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/requests/sessions.py", line 596, in send
                                                    r = adapter.send(request, **kwargs)
                                                  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/requests/adapters.py", line 487, in send
                                                    raise ConnectionError(e, request=request)
                                                ConnectionError: HTTPConnectionPool(host='localhost', port=17123): Max retries exceeded with url: /api/v1/series?api_key=<PLAIN_TEXT_API_KEY_REDACTED>(Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f5268dd8a50>: Failed to establish a new connection: [Errno 111] Connection refused',))
Sep 28 23:21:31 HOSTNAME systemd[1]: Starting "Datadog Agent"...
Sep 28 23:21:31 HOSTNAME systemd[1]: Started "Datadog Agent".
```

**Describe the results you expected:**
Would expect to not see my API key in plain text.

**Comment by rnkaufman** [comment]
Looks like this got fixed recently: https://github.com/DataDog/dd-agent/pull/3537

**Comment by hush-hush** [comment]
Thank you a lot @pjvelez for reporting the issue and sorry for the late feedback. It should indeed be fix since `5.18`.
I would advise updating to `5.18` and changing your Datadog API key.

I'm going to close this issue but feel free to reopen it if needed.
